### PR TITLE
[FEATURE] Extract validation result failures

### PR DESCRIPTION
**Pull request by BenGale93**
Changes proposed in this pull request:
- Added get_failed method onto ExpectationSuiteValidationResult.
- Dealt with the TODO found in data_context.py

### Note

To create the new ExpectationSuiteValidationResult in get_failed I recalculated the statistics attribute using a function from data_asset, namely the _calc_validation_statistics function. Ideally I would just directly use this function but given it is marked as private I chose not to.

I feel though that this function could be an internal method of ExpectationSuiteValidationResult which computes the statistics during initialisation, based on what results are passed in. That way the statistics are never out of line with the results found in the instance. Happy to open a pull request to implement this if you would like?

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.

### CODE REVIEW

**Comment by talagluck** [comment]
Thanks for submitting, @BenGale93! We will review and be in touch in the next couple of days.

File: great_expectations/core/expectation_validation_result.py [hunk 1]
```diff
@@ -371,6 +371,33 @@ def get_metric(self, metric_name, **kwargs):
             )
         )
 
+    def get_failed(self):
```
Comment (cdkini) [inline great_expectations/core/expectation_validation_result.py]: Could we perhaps rename this to `get_failed_validation_results` just to be very explicit?

Additionally, could we add `ExpectationSuiteValidationResult` as a return type? We're trying to make a more conscious effort to type hint so any additions are greatly appreciated!

**Review by cdkini — CHANGES_REQUESTED** [review]
Requesting a very minor change but this all looks great to me @BenGale93 ! Once you make the minor updates, I'll approve and merge

File: great_expectations/data_context/data_context.py [hunk 2]
```diff
@@ -2428,15 +2428,7 @@ def get_validation_result(
...
-        # TODO: This should be a convenience method of ValidationResultSuite
-        if failed_only:
-            failed_results_list = [
-                result for result in results_dict.results if not result.success
-            ]
-            results_dict.results = failed_results_list
-            return results_dict
-        else:
-            return results_dict
+        return results_dict.get_failed() if failed_only else results_dict
```
Comment (cdkini) [inline great_expectations/data_context/data_context.py]: I think this usage needs to be fixed before we get it all working?

Comment (cdkini) [inline great_expectations/data_context/data_context.py]:
```suggestion
        return results_dict.get_failed_validation_results() if failed_only else results_dict
```

**Review by cdkini — APPROVED** [review]

**Comment by cdkini** [comment]
@BenGale93 decided to just make the quick update. As long as tests pass this time around, the changes will be merged. Thanks so much!

**Pull request merged**